### ISSUE TITLE
Inconsistent Management API Responses

### ISSUE THREAD

**Issue by tustvold** [influxdata/influxdb_iox#3247]
**Problem**

Some operations within the management API return a longrunning operations handle as soon as the task has been "accepted":

* ClosePartitionChunk
* WipePreservedCatalog

Others don't return a longrunning operation at all:

* SkipReplay
* PersistPartition
* DropPartition
* UnloadPartitionChunk

This makes for an inconsistent API which is not ideal.

The major reason to favor eagerly returning a longrunning operation handle is it allows setting tight request [deadlines](https://grpc.io/blog/deadlines/), the major downside is it complicates client implementations. 

**Proposal**

There are broadly 3 categories of operation:

1. Many Minutes - WipePreservedCatalog
2. Seconds to minutes - PersistPartition, ClosePartitionChunk
3. Milliseconds to Seconds - SkipReplay, UnloadPartitionChunk, DropPartition

I think creating and returning longrunning task handles for category 3 is unnecessary, similarly I think not eagerly returning a handle for category 1 is not a good idea. I'm not sure what the correct choice for category 2 is, perhaps create longrunning operation internally, but use synchronous RPC with the client?

**Comment by crepererum** [comment]
A major difference between category 2 and 3 is that the former involves object store IO. I think if it's avoidable we should never block a request on external IO (query request being a notable exception to this rule).

FWIW I would also favor a bit more implementation complexity if it results in a more consistent API design. This design not only affects that gRPC interface but also the Rust client as well as the CLI. So I would also like to see category 3 to return job handles eagerly, but that's not a too strong opinion.

**Comment by alamb** [comment]
I agree with @crepererum. So specifically, a counter proposal is  that all APIs return long running all categories you refer to and adding code to the Rust client that waits for the operation to complete successfully that is convenient to use.

> major downside is it complicates client implementations.

FWIW I am not sure it complicates client implementations (as the client already has to handle long running operations for certain APIs)

**Comment by tustvold** [comment]
No objections from me to always eagerly returning an operations handle :+1:
